data/transactions/transaction.go

```diff
@@ -93,6 +93,12 @@ type Header struct {
 	// the LastValid round passes.  While this transaction possesses the
 	// lease, no other transaction specifying this lease can be confirmed.
 	Lease [32]byte `codec:"lx"`
+
+	// RekeyTo, if nonzero, sets the sender's SpendingKey to the given address
+	// If the RekeyTo address is the sender's actual address, the SpendingKey is set to zero
+	// This allows "re-keying" a long-lived account -- rotating the signing key, changing
+	// membership of a multisig account, etc.
+	RekeyTo basics.Address `codec:"rekey"`
```

Comment:
Overall, this PR looks great!

Minor suggestion/comment: rekeying is a major operation, but this allows any transaction to also rekey.  I wonder if it's worth defining a new txtype for rekey, or at least tacking it onto keyreg txtypes (so that, for example, a payment can never rekey)?

Reply:
Yeah, one alternative design was to have a separate rekeying transaction, and then have transactions that wanted to do X and rekey atomically bundled in a group transaction.

The assumption here was that a small proportion of high-security accounts will want to rekey on every transaction and that for these accounts, it would be marginally more convenient to have everything done in one transaction. (A TEAL program can enforce that a particular transaction doesn't rekey by specifying that the `RekeyTo` field is zero.)

Reply:
Ah, yeah, that seems like an interesting use case.  (Of course doable with tx groups too, but not obvious which plan is better overall.)